kernel/src/table.rs

```diff
@@ -46,6 +83,46 @@ impl Table {
     }
 }
 
+#[derive(Debug)]
+enum UriType {
+    LocalPath(PathBuf),
+    Url(Url),
+}
+
+/// Utility function to figure out whether string representation of the path is either local path or
+/// some kind or URL.
+///
+/// Will return an error if the path is not valid.
+fn resolve_uri_type(table_uri: impl AsRef<str>) -> DeltaResult<UriType> {
+    let table_uri = table_uri.as_ref();
+    let known_schemes = [
+        "file", "memory", "s3", "s3a", "az", "adl", "azure", "abfs", "abfss", "gs", "http", "https",
+    ]; // hard-code for now, TODO: pull these out of object_store
```

Comment:
This is repeating one of the more annoying mistakes we made early in delta-rs. The approach now allows for registering file scheme handlers, which allows for third party crates to bring their own `ObjectStore` compatible implementations and for everything else to just work.

The `register_handler` code isn't my favorite in delta-rs but it is allowing for third parties to create storage implementations and the `deltalake::open_table()`  style calls will still just work.

Reply:
Good to know thanks :)

If I understand your comment correctly, we should just not validate at that point and either we find some compatible object store when we instantiate the client, or we don't, and the error can happen there?